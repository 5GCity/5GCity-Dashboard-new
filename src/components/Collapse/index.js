/**
 * Collapse Component
 * Please write a description
 *
 */
import React from 'react'
import styled from 'styled-components'
import { Collapse } from 'element-react'

export default ({ children, ...props }) => (
  <Wrapper>
    <Collapse {...props}>
      {children && children.map((child, i) =>
        <Collapse.Item
          key={i}
          title={child.props && createTitle(child.props)}
          name={child.props && child.props.name}
      >
          {child}
        </Collapse.Item>
    )}
    </Collapse>
  </Wrapper>
)

const createTitle = props => {
  return (
    <React.Fragment>
      {props.title && props.title}
      {props.onClick &&
      <Icon onClick={(e) => {
        e.stopPropagation()
        props.onClick('AddNew')
      }
      } >
        {props.icon &&
        props.icon
      }
      </Icon>
  }
    </React.Fragment>
  )
}

const Wrapper = styled.div`
  .el-collapse {
    border: 0px solid transparent;
  }
  .el-collapse-item__header {
    height: 32px;
    line-height: 32px;
    padding-left: 10px;
    background-color: #324148;
    font-family: ${({theme}) => theme.fontFamily};
    color: ${({theme}) => theme.secondaryColor};
    cursor: pointer;
    border-bottom: none;
    font-size: 14px;
    border-radius: none;
  }
  .el-collapse-item__wrap{
    background-color: transparent;
    border-bottom: none;
    color: white;
  }
  .el-collapse-item__content{
    color: white;
    padding: 5px;
  }
`
const Icon = styled.span`
  float: right;
  margin-right: 9px;
`
