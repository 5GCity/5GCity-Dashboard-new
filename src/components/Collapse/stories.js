/**
 * Collapse Component Stories
 * Please write a description or remove this line
 *
 * @author Guilherme Patriarca <gpatiarca@ubiwhere.com>
 */
import React from 'react'
import { storiesOf, action } from '@storybook/react'

import Collapse from './index'
import Catalogue from 'components/Catalogue'
import { CirclePlus } from 'components/Icons'

const CollapseOne = () => (
  <Collapse value={'favorite'} onChange={action('Collapse on change')}>
    <p title={'My catalogue'}>Catalogue</p>
    <p
      title={'Favorite'}
      name={'favorite'}
      onClick={action('Click Catalogue')}
      icon={<CirclePlus />}
    >
      <Catalogue
        version={'3.0.1'}
        type={'VNF'}
        colortext='blue'
        circlefill='yellow'
        name={'other mec with long name'}
        onClick={action('Click Catalogue')}
      />
      <Catalogue
        version={'1.0.1'}
        type={'VNF'}
        colortext='blue'
        circlefill='yellow'
        name={'other name'}
        onClick={action('Click Catalogue')}
      />
    </p>
  </Collapse>
)

storiesOf('Collapse', module)
  .add('default', CollapseOne)
