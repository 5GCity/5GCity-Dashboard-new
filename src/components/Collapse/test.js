/**
 * Collapse Component Tests
 *
 * @author Guilherme Patriarca <gpatiarca@ubiwhere.com>
 */
import React from 'react'
import renderer from 'react-test-renderer'
import Collapse from './index'
import 'jest-styled-components'

it('renders correctly', () => {
  const tree = renderer
    .create(<Collapse />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
