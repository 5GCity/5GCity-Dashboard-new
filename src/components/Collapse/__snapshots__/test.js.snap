// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 .el-collapse {
  border: 0px solid transparent;
}

.c0 .el-collapse-item__header {
  height: 32px;
  line-height: 32px;
  padding-left: 10px;
  background-color: #324148;
  cursor: pointer;
  border-bottom: none;
  font-size: 14px;
  border-radius: none;
}

.c0 .el-collapse-item__wrap {
  background-color: transparent;
  border-bottom: none;
  color: white;
}

.c0 .el-collapse-item__content {
  color: white;
  padding: 5px;
}

<div
  className="c0"
>
  <div
    className="el-collapse"
  />
</div>
`;
